storage/engine/mvcc_test.go

```diff
@@ -1236,6 +1236,43 @@ func TestMVCCResolveTxn(t *testing.T) {
 	}
 }
 
+// TestMVCCConditionalPutOldTimestamp tests a case where a conditional
+// put with an older timestamp happens after a put with a newer timestamp.
+func TestMVCCConditionalPutOldTimestamp(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	engine := createTestEngine()
+	defer engine.Close()
+
+	err := MVCCPut(engine, nil, testKey1, makeTS(1, 0), value1, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = MVCCPut(engine, nil, testKey1, makeTS(3, 0), value2, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// QUESTION(kkaneda): Should this fail with WriteTooOldError?
```

Comment:
Yeah, these errors look backwards. If you decomposed the ConditionalPut into a transaction with a Get and Put, the first would match the existing value and then fail with a WriteTooOld error, while the second would look for the value at t=2 and get ConditionFailedError. (Actually, the timestamps in this test should result in ReadWithinUncertaintyInterval, but with more space between them you'd still get WriteTooOld and ConditionFailed).